csrc/quantization/fp8/common.cuh

```diff
@@ -7,18 +7,47 @@
 
 #ifndef USE_ROCM
   #include <c10/util/Float8_e4m3fn.h>
-using FP8_TYPE = c10::Float8_e4m3fn;
-C10_HOST_DEVICE constexpr auto FP8_E4M3_MAX =
-    std::numeric_limits<FP8_TYPE>::max();
 #else
+  #include <ATen/hip/HIPContext.h>
+  #include <c10/util/Float8_e4m3fn.h>
   #include <c10/util/Float8_e4m3fnuz.h>
   #include "amd/quant_utils.cuh"
-using FP8_TYPE = c10::Float8_e4m3fnuz;
-// Using the default max value from pytorch (240.0) will cause accuracy
-// issue when running dynamic quantization. Here use 224.0f for rocm.
-constexpr auto FP8_E4M3_MAX = 224.0f;
 #endif
-constexpr static auto kFp8Type = c10::CppTypeToScalarType<FP8_TYPE>::value;
+
+static bool is_fp8_ocp() {
```

Comment:
Could you add a brief comment about this method? I'd mention it checks device properties and .

It would also be good to check the property directly instead of querying all device properties, which is significantly slower (in my experience on CUDA, not sure if the same applies on AMD)

Reply:
I will add a comment, yes.

Would it ease your mind to know that `at::cuda::getCurrentDeviceProperties()` is cached internally by pytorch?

See https://github.com/pytorch/pytorch/blob/ed9c8a5d136cae1dac46c7700b83d3e07830cff8/aten/src/ATen/cuda/CUDAContext.cpp#L17-L60 for the implementation.